padmy/utils.py

```
@@ -317,9 +410,230 @@ async def get_conn(pool: asyncpg.Pool, fn: Callable):
+        try:
+            yield
+        finally:
+            # Restore original environment
+            os.environ.clear()
+            os.environ.update(old_env)
+
+    @contextlib.asynccontextmanager
+    async def get_conn(self, database: str | None = None, timeout: int = 5):
+        _database = database if database is not None else self.database
+        if _database is None:
+            raise ValueError("Either database parameter or PGConnectionInfo.database must be set")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)